router_api.go

```diff
@@ -15,8 +31,17 @@ func newAPIHandler(rout *Router) http.Handler {
 			w.WriteHeader(http.StatusMethodNotAllowed)
 			return
 		}
-
-		rout.ReloadRoutes()
+		// Send a message to the reload goroutine, which will start a new timeout
+		// before reloading, or do nothing if one is already in progress.
+		select {
+		case reloadChan <- true:
+			logInfo("router: reload triggered")
+			w.WriteHeader(http.StatusAccepted)
+			w.Write([]byte("Reload triggered"))
+		default:
+			logInfo("router: reload already in progress")
+			w.Write([]byte("Reload already in progress"))
+		}
```

Comment:
I'm not sure there's a race here: before the sleep starts the value should have been read from the channel, so the channel will be already be empty and a subsequent call to send a value to the channel will work.  The channel can't then be cleared again until that new value is read.

I think this has be true, because every value read from the channel triggers a reload, and values are removed from the channel atomically compared to checking whether the channel is empty. (Though this atomicity isn't actually required, you just need to be sure that the values are removed from the channel before the reload start.)